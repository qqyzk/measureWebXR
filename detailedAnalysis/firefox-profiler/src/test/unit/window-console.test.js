/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

// @flow
import {
  addDataToWindowObject,
  logFriendlyPreamble,
} from '../../utils/window-console';
import { storeWithSimpleProfile } from '../fixtures/stores';

describe('console-accessible values on the window object', function () {
  // Coerce the window into a generic object, as these values aren't defined
  // in the flow type definition.

  it('does not have the values initially', function () {
    expect((window: any).profile).toBeUndefined();
    expect((window: any).filteredProfile).toBeUndefined();
    expect((window: any).callTree).toBeUndefined();
  });

  it('adds values to the console', function () {
    const store = storeWithSimpleProfile();
    const target = {};
    addDataToWindowObject(store.getState, store.dispatch, target);
    expect(target.profile).toBeTruthy();
    expect(target.filteredThread).toBeTruthy();
    expect(target.callTree).toBeTruthy();
  });

  it('logs a friendly message', function () {
    const log = console.log;
    (console: any).log = jest.fn();
    logFriendlyPreamble();
    expect(console.log.mock.calls.length).toEqual(2);
    expect(console.log.mock.calls).toMatchSnapshot();
    (console: any).log = log;
  });
});
