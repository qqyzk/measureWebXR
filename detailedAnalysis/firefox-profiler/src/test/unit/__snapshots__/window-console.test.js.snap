// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`console-accessible values on the window object logs a friendly message 1`] = `
Array [
  Array [
    "%c  __ _           __                                  
 / _(_)         / _|                                 
| |_ _ _ __ ___| |_ _____  __                        
|  _| | '__/ _ \\\\  _/ _ \\\\ \\\\/ /                        
| | | | | |  __/ || (_) >  <       __ _ _            
|_| |_|_|  \\\\___|_| \\\\___/_/\\\\_\\\\     / _(_) |           
    ,.       ,.   _ __  _ __ ___ | |_ _| | ___ _ _   
    | \\\\     / |  | '_ \\\\| '__/ _ \\\\|  _| | |/ _ \\\\ '_|  
    |/ \\\\ _ / \\\\|  | |_) | | | (_) | | | | |  __/ |    
    |         |  | .__/|_|  \\\\___/|_| |_|_|\\\\___|_|    
    /  -    - \\\\  |_|                                     
  ,-    V__V   -.                                     
 -=  __-  * - .,=-                                    
  \`\\\\_    -   _/                                       
      \`-----'                                         ",
    "font-family: Menlo, monospace;",
  ],
  Array [
    "%cThe following profiler information is available via the console:%c

%cwindow.profile%c - The currently loaded profile
%cwindow.filteredThread%c - The current filtered thread
%cwindow.filteredMarkers%c - The current filtered and processed markers
%cwindow.callTree%c - The call tree of the current filtered thread
%cwindow.getState%c - The function that returns the current Redux state.
%cwindow.selectors%c - All the selectors that are used to get data from the Redux state.
%cwindow.dispatch%c - The function to dispatch a Redux action to change the state.
%cwindow.actions%c - All the actions that can be dispatched to change the state.
%cwindow.experimental%c - The object that holds flags of all the experimental features.
%cwindow.togglePseudoLocalization%c - Enable pseudo localizations by passing \\"accented\\" or \\"bidi\\" to this function, or disable using no parameters.
%cwindow.toggleTimelineType%c - Toggle timeline graph type by passing \\"cpu-category\\", \\"category\\", or \\"stack\\".

The profile format is documented here:
%chttps://github.com/firefox-devtools/profiler/blob/main/docs-developer/processed-profile-format.md%c

The CallTree class's source code is available here:
%chttps://github.com/firefox-devtools/profiler/blob/main/src/profile-logic/call-tree.js%c",
    "font-size: 130%;",
    "",
    "font-weight: bold;",
    "",
    "font-weight: bold;",
    "",
    "font-weight: bold;",
    "",
    "font-weight: bold;",
    "",
    "font-weight: bold;",
    "",
    "font-weight: bold;",
    "",
    "font-weight: bold;",
    "",
    "font-weight: bold;",
    "",
    "font-weight: bold;",
    "",
    "font-weight: bold;",
    "",
    "font-weight: bold;",
    "",
    "font-style: italic; text-decoration: underline;",
    "",
    "font-style: italic; text-decoration: underline;",
    "",
  ],
]
`;
